examples/local_hugging_face_server.py

```diff
@@ -0,0 +1,89 @@
+"""Example server definition for deploying Huggingface models to a local endpoint.
+Sets up the /completions endpoint which (broadly) conforms to what's described
+in the link below.
+https://alexatallah.notion.site/RFC-LLM-API-Standard-c8f15d24bd2f4ab98b656f08cdc1c4fb#e891724fc18f4524a970aca2379e8985
+Main thing to note is the `import_and_return` functions which download the
+Huggingface model and returns the necessary components to use it.
+This connector function is stored in the `CONNECTED_MODELS` dictionary.
+"""
+from flask import Flask, jsonify, request
+from transformers import AutoTokenizer, AutoModelForSeq2SeqLM
+from transformers import BertTokenizer, BertForSequenceClassification
+from transformers import pipeline
+
+app = Flask(__name__)
+
+
+"""Connector functions."""
+
+def import_and_return_tk_instruct_3b_def():
+    """Example import function for Huggingface model. Returns model, tokenizer,
+    and code for running the model."""
+    tokenizer = AutoTokenizer.from_pretrained("allenai/tk-instruct-3b-def")
+    model = AutoModelForSeq2SeqLM.from_pretrained("allenai/tk-instruct-3b-def")
+
+    def model_fn(model, string, tokenizer):
+        input_ids = tokenizer.encode(string, return_tensors="pt")
+        output = model.generate(input_ids, max_length=10)
+        output = tokenizer.decode(output[0], skip_special_tokens=True)
+        return output
+
+    return model, tokenizer, model_fn
+
+
+def import_and_return_finbert_tone():
+    """Imports https://huggingface.co/yiyanghkust/finbert-tone."""
+    finbert = BertForSequenceClassification.from_pretrained('yiyanghkust/finbert-tone', num_labels=3)
+    tokenizer = BertTokenizer.from_pretrained('yiyanghkust/finbert-tone')
+
+    def model_fn(model, string, tokenizer):
+        nlp = pipeline("sentiment-analysis", model=model, tokenizer=tokenizer)
+        return nlp([string])
+
+    return finbert, tokenizer, model_fn
+
+
+
+CONNECTED_MODELS = {
+    "allenai/tk-instruct-3b-def": import_and_return_tk_instruct_3b_def,
+    "yiyanghkust/finbert-tone": import_and_return_finbert_tone,
+}
+
+
+"""Endpoint."""
+
+@app.route("/completions", methods=["POST"])
+def completions():
+    """The completions endpoint."""
+    # Get the request data.
+    data = request.get_json()
+    model = data.get("model", "yiyanghkust/finbert-tone")
+    prompt = data["prompt"]
```

Comment:
since this example is restrictive and doesn't work with the full api (chatml, streaming etc) can you include an index.html file with a sample webapp that calls it through window?

Reply:
Yep, added an `index.html` file. That was surprisingly easy!